### ISSUE TITLE
[FEATURE] Enable Hover on Label or Title

### ISSUE THREAD

**Issue by ncclementi** [bokeh/bokeh#11389]
Overall problem: 
Trying to have an informative description (HTML text) about a plot to pop up on the plot when hovering over a label 

**Is your feature request related to a problem? Please describe.**
I'm currently trying to have a Label fixed on the top right corner of a plot (displaying a question mark), that when I hover over it, it displays a description about the plot. 
I know that the toolbar `?` can be use to link to a description of the plot, but I want the description to be on the plot not on an external link. But currently I can't link a HoverTool to a Label. 

```python
from bokeh.plotting import figure, show
from bokeh.models import Label,  HoverTool

x = [1, 2, 3, 4, 5, 6]
y = [5, 3, 7, 7, 1, 2]

p = figure(plot_width=600, plot_height=600)
p.line(x=x, y=y)

my_label = Label(x=p.plot_width - 70, y=p.plot_height-50, x_units="screen", y_units="screen", text="?")

p.add_layout(my_label)

hover = HoverTool(
            point_policy="follow_mouse",
            tooltips="""Your <a href="https://en.wikipedia.org/wiki/HTML">HTML</a>-supported text is 
initialized with the <b>text</b> argument.  The remaining div arguments are <b>width</b> and <b>height</b>. For
this example, those values are <i>200</i> and <i>100</i>, respectively.""",
            renderers=[my_label],
        )

p.add_tools(hover)

show(p)
```
```
ValueError: failed to validate HoverTool(id='4921', ...).renderers: expected an element of either Auto or List(Instance(DataRenderer)), got [Label(id='4920', ...)]
```

**Describe the solution you'd like**
I would like to be able to have a question mark displaying on the top right corner of a plot ( using `x_units` and `y_units`) that when I hover over it it provide a description of the plot in HTML text. 

**Describe alternatives you've considered**

- I though of displaying a glyph or ImageURL below the label and render the hover on it, but none of the glyphs provide `x_units` or `y_units` which is a problem, given that the plots are on a dashboard that change sizes and I want the Label/glyph to always be on the same place (top right corner of the plot)

**Additional context**
-(1) I was able to have a button that when clicked it display HTML but neither the button or the text are in the plot, but on the side (see image below). If there is a chance of having the button right next to the title TESTING (on the right side not left) and the HTML text to display over the button when clicked (not outside the plot area), this could be an alternative. 
- (2) Another possible option would be to be able to have a hover tooltip to display on the Title object, but hovers currently don't display on Title.

code for (1)
```python

from bokeh.plotting import figure, show, output_notebook
from bokeh.models import Label, Button, CustomJS, Title, Div, HoverTool
from bokeh.layouts import column, row

p = figure()

title_test = Title(text="TESTING", text_font_style="italic", align ="left", text_font_size = "25px")

p.line(x=x, y=y)

p.add_layout(title_test, "above")


my_label = Div(text="""Your <a href="https://en.wikipedia.org/wiki/HTML">HTML</a>-supported text is 
initialized with the <b>text</b> argument.  The remaining div arguments are <b>width</b> and <b>height</b>. For
this example, those values are <i>200</i> and <i>100</i>, respectively.""",
width=200, height=100)


my_label.visible = False


callback = CustomJS(args=dict(my_label=my_label), code="""

    // if it's visible now, make it Invisible...

    if (my_label.visible == true) 

      my_label.visible = false;

    // otherwise, make it visible.

    else 

      my_label.visible = true;


""")


button = Button(label="info")

button.js_on_click(callback)

show(row(column(button, width=50), p, my_label))
```
![Screen Shot 2021-06-29 at 3 32 22 PM](https://user-images.githubusercontent.com/7526622/123856555-3c934580-d8ef-11eb-9f14-11919b924e5b.png)

**Comment by mattpap** [comment]
Cross-referencing issue #8727, as the required changes on the implementation level are pretty much the same, just touching a different area of the API.

**Comment by bryevdv** [comment]
Just chiming in that if I understand your comment, we agree that this should be imagined more in terms of tooltip a new capability defined on the objects themselves, and not in terms of extending the existing hovertool to include non-glyphs.

**Comment by ghomem** [comment]
I am also interested in this feature, specifically

> to have a Label fixed on the top right corner of a plot (displaying a question mark), that when I hover over it, it displays a description about the plot.

would very nice for allowing the plots to explain themselves, on hover, without cluttering an entire dashboard with permanent explanations. I would guess this is useful in general for users.

Thank you :-)

**Comment by ghomem** [comment]
If I understand correctly this would be something like what is seen in the attached images, but applied to the plot title
![a6TyU](https://user-images.githubusercontent.com/3352900/141431603-c05f07c5-1efd-4318-8309-0b71c93a8a90.png)
![402c93bec364ed7e309eb01bdb440e2e84bd367a](https://user-images.githubusercontent.com/3352900/141431624-2a45fd39-f997-4558-bd30-6506eda1a01a.png)
.

**Comment by ncclementi** [comment]
@ghomem Yes, that is what I have in mind when I requested the feature but I'm not sure when would be implemented.

**Comment by bryevdv** [comment]
Unfortunately it's just not a trivial ask. Titles are drawn on a canvas, they are  not DOM elements, so it's not possible to leverage DOM tooltips. And they are not driven by CDS data sources, so we can't just reuse all the existing hover tooltip machinery built for glyphs, either. So it will have to be something completely new and separate.

**Comment by ghomem** [comment]
Thanks for looking into this @bryevdv . Would it be easier if the tooltip was associated with the plot, rather than the title? Perhaps it is the plot itself that needs an extra field that makes it self explanatory. I mean, the title of the plot is a condensed explanation of the plot, whereas this new field would be an expanded explanation, only visible upon user click on an "information" icon. No idea if this makes the implementation easier but perhaps makes the concept more logical.
